### PR TITLE
Fixed issue with <br> elements

### PR DESCRIPTION
**Pull request by SetTrend**
# Synopsis
Prism ignored `<br>` elements and removed them from the code. This behaviour was based on the fact that `Prism.highlightElement()` used `element.textContent`, which expunges `<br>` elements (because they are no text nodes).

I replaced the use of `element.textContent` with `element.innerText` (which correctly produces a node's text content whitespace) and removed a call to `replace()` that was removing newlines from the tokens.

So, this commit retains the new-lines correctly.

See issue #1764 .
<hr/>

### Remarks
I only tested my personal use case.

### CODE REVIEW

File: components/prism-core.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ var _ = _self.Prism = {
 			} else if (_.util.type(tokens) === 'Array') {
 				return tokens.map(_.util.encode);
 			} else {
-				return tokens.replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/\u00a0/g, ' ');
```
Comment (RunDevelopment) [inline components/prism-core.js]: Why did you remove this?

Comment (SetTrend) [reply components/prism-core.js]: `<pre>` is supposed to retain newline characters, so it doesn't make sense to remove them (in the original code, see merge request).

Comment (SetTrend) [reply components/prism-core.js]: This line caused text tokens to be altered from `\nTest\n` to ` Test `.

So what was ...

```javascript
var i;

Test

var j;
```

... in the original `<code>` element became ...

```javascript
var i; Test var j;
```

Comment (SetTrend) [reply components/prism-core.js]: See my sample HTML file in issue #1764.

Check the result when you alter the code content to:

```javascript
var i = 1;

Test

function()
{}
```

Comment (RunDevelopment) [reply components/prism-core.js]: This line only escapes text to be valid HTML, so that e.g. the code `<p>` is not interpreted as a tag but as text.

I'm asking why you removed the removal of No-break spaces. I don't see how this relates to new lines.
(TBH, I don't know why we remove NBSPs in the first place)

Comment (SetTrend) [reply components/prism-core.js]: D'oh! You are right. I misread the `\u00a0` as `\u000a`.

The Firefox debugger I'm using is not very stable. While debugging it turned out that the debugger stripped the new-lines from my test when I stepped over that function. I then examined that function and misread the literal value.

Yet, it's like you wrote: There is no need to escape the non-breaking space. I can remove the change, though, if you prefer.

Comment (RunDevelopment) [reply components/prism-core.js]: Yes, please remove this change.
It's better to keep things simple.

**Comment by ExE-Boss** [comment]
I think a better solution would be a hook that would auto&#x2011;convert `<br>` elements into `\n`.

**Comment by SetTrend** [comment]
As @RunDevelopment correctly mentioned in [his reply to #1764](/PrismJS/prism/issues/1764#issuecomment-467501221), `innerText` also adds `\n` for non-nested block elements, mimicking what the user sees. I guess the complex calculation for what's supposed to add a `\n` character is what's making `innerText` slower than `textContent`.

**Comment by SetTrend** [comment]
As I wrote yesterday in #1764, I'm now going to close this pull-request.

I'd be interested, though, to learn how many users preferred the current behaviour against the suggested behaviour. I believe the current behaviour is odd, but I'm open to opinions, though.

**Comment by RunDevelopment** [comment]
I'm going to make a new issue for that.
